src/ripple/consensus/Validations.h

```diff
@@ -966,6 +978,42 @@ class Validations
 
         adaptor_.flush(std::move(flushed));
     }
+
+    /** Return quantity of lagging proposers, and remove offline proposers
+     *  for purposes of evaluating whether to pause.
+     *
+     *  Laggards are the trusted proposers whose sequence number is lower
+     *  than the sequence number from which our current pending proposal
+     *  is based. Proposers are considered offline from whom we have not
+     *  received a validation for awhile.
+     *
+     *  Note: the trusted flag is not used in this evaluation because it's made
+     *  redundant by checking the list of proposers.
+     *
+     * @param seq Our current sequence number.
+     * @param trustedNodes Public keys of trusted proposers.
+     * @return Quantity of laggards.
+     */
+    std::size_t
+    laggards(Seq const seq, hash_set<NodeKey>& trustedNodes)
```

Comment:
fixed

Reply:
Same nit/question on the latest code---looks like this was just renamed to `trustedKeys`. How does it differ from the `trusted` flag of each validation?

Reply:
I use the whole set of trustedKeys to determine how many validators are offline based on not having received a recent validation. Since I'm checking validations against trustedKeys already, also checking for the trusted flag becomes redundant. I do not believe that they would differ, except perhaps in a very small window of time when the validator list changes.